### PR TITLE
Add mssql support via sqlpackage

### PR DESCRIPTION
**Pull request by 1k-off**
Added Microsoft SQL database support with sqlpackage utility.
Updated Dockerfile (add necessary dependencies).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +43,7 @@ https://gobackup.github.io
 - Redis - `mode: sync/copy`
 - MongoDB
 - SQLite
+- MSSQL
```
Comment (huacnlee) [inline README.md:46]: Rename to `Microsoft SQL Server`

File: database/mssql_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package database
+
+import (
+	// "github.com/spf13/viper"
```
Comment (huacnlee) [inline database/mssql_test.go:4]: Remove this line

File: database/mssql.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		if len(db.port) > 0 {
+			return "/SourceServerName:" + db.host + "," + db.port
+		}
+		return "/SourceServerName:" + db.host + "," + "1433"
+	}
+	if len(db.port) > 0 {
+		return "/SourceServerName:127.0.0.1" + "," + db.port
+	}
+	return "/SourceServerName:127.0.0.1,1433"
+}
```
Comment (huacnlee) [inline database/mssql.go:90]:
```go
func (db *MSSQL) connectivityOptions() string {
	var host = db.host
	var port = db.port

	if len(host) == 0 {
		host = "127.0.0.1"
	}
	if len(port) == 0 {
		port = "1433"
	}

	return "/SourceServerName:" + db.host + "," + db.port
}
```

File: database/mssql.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	return "/SourceDatabaseName:" + db.database
+}
+
+func (db *MSSQL) credentialOptions() string {
+	opts := []string{}
+	if len(db.username) > 0 {
+		opts = append(opts, "/SourceUser:"+db.username)
+	}
+	if len(db.password) > 0 {
+		opts = append(opts, `/SourcePassword:`+db.password)
```
Comment (huacnlee) [inline database/mssql.go:74]: ``` `/SourcePassword:` ``` to `"/SourcePassword:"`

File: Dockerfile [hunk 5]
```diff
@@ -24,7 +24,21 @@ RUN apk add \
   && \
   rm -rf /var/cache/apk/*
 
+WORKDIR /tmp
+RUN apk update && \
```
Comment (huacnlee) [inline Dockerfile:28]: Merge them to the before `RUN` script, with one line.

Comment (huacnlee) [reply Dockerfile:28]: Or just move the `libstdc++ gcompat icu`, and keep the `sqlpackage` install in split.

Comment (1k-off) [reply Dockerfile:28]: I'll keep mssql installation in the separate layer. Thanks.

**Comment by huacnlee** [comment]
image


Test has failed.

**Comment by 1k-off** [comment]
> image
> Test has failed.

Yep, we can't use empty host. I will add some logic tonight.

**Comment by 1k-off** [comment]
Fixed now. Sorry for inattention.

**Pull request merged**